chia/data_layer/data_layer.py

```diff
@@ -621,6 +624,30 @@ async def fetch_and_validate(self, store_id: bytes32) -> None:
             except Exception as e:
                 self.log.warning(f"Exception while downloading files for {store_id}: {e} {traceback.format_exc()}.")
 
+        # if there aren't any servers then don't try to write the full tree
+        if not success and len(servers_info) > 0:
```

Comment:
Should we just entirely short circuit with a `return` above if there are no servers and skip the loop etc?